test-validator/src/lib.rs

```diff
@@ -358,9 +358,39 @@ impl TestValidatorGenesis {
         socket_addr_space: SocketAddrSpace,
     ) -> (TestValidator, Keypair) {
         let mint_keypair = Keypair::new();
-        TestValidator::start(mint_keypair.pubkey(), self, socket_addr_space)
-            .map(|test_validator| (test_validator, mint_keypair))
-            .expect("Test validator failed to start")
+        match TestValidator::start(mint_keypair.pubkey(), self, socket_addr_space) {
+            Ok(test_validator) => {
+                let runtime = tokio::runtime::Builder::new_current_thread()
+                    .enable_io()
+                    .enable_time()
+                    .build()
+                    .unwrap();
+                runtime.block_on(test_validator.wait_for_nonzero_fees());
+                (test_validator, mint_keypair)
+            }
+            Err(err) => panic!("Test validator failed to start: {}", err),
+        }
+    }
+
+    pub async fn nonblocking_start(&self) -> (TestValidator, Keypair) {
```

Comment:
I don't love the name `nonblocking_start()` here :-/

Reply:
Is `start_nonblocking()` any better?